### PR TITLE
User unsubscribe

### PR DESCRIPTION
**Pull request by schwesig**
add webhook delete
closes https://github.com/open-services-group/peribolos-as-a-service/issues/8

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -72,6 +71,18 @@ export = (app: Probot) => {
     };
     await k8sApi.createNamespacedSecret(namespace,secret);
   });
+
+  // Respond to the GitHub app installation deleted
+  app.on("installation.deleted", async (context) =>{
+
+    // ? user story: delete repo or just the secret ?
```
Comment (schwesig) [inline src/index.ts:78]: is deleting just the secret enough (following the use case, yes)? Do we need more clean up?
[User unsubscribe](https://github.com/open-services-group/peribolos-as-a-service/blob/fab17cbeb13e9ec6d95ffed4b7952089558583cd/architecture.md#user-unsubscribe)

Comment (schwesig) [reply src/index.ts:78]: for now it is enough.
acting on an repo and getting the secret can become a problem.
but basic idea would be:
Run a task deleting the peribolos.yaml from .github repo and opening a PR.

File: src/index.ts [hunk 2]
```diff
@@ -72,6 +72,18 @@ export = (app: Probot) => {
...
+    // ? do we have to take care of repo
+    // TODO: check if secret exists and catch errors
+        
+    var name = "peribolos-"+context.payload.installation.id;
+    
+    await k8sApi.deleteNamespacedSecret(name, namespace);
```
Comment (harshad16) [inline src/index.ts:85]:
```suggestion
    await k8sApi.deleteNamespacedSecret(name, k8snamespace);
```
can we make the variable as suggested, this change would be introduced on other places via: https://github.com/open-services-group/peribolos-as-a-service/pull/21

**Review by harshad16 — CHANGES_REQUESTED** [review]
lgtm
just need one change and pre-commit run

File: src/index.ts [hunk 3]
```diff
@@ -72,6 +72,18 @@ export = (app: Probot) => {
     };
     await k8sApi.createNamespacedSecret(namespace,secret);
   });
+
+  // Respond to the GitHub app installation deleted
+  app.on("installation.deleted", async (context) =>{
+
+    // ? user story: delete repo or just the secret ?
+    // ? do we have to take care of repo
+    // TODO: check if secret exists and catch errors
```
Comment (tumido) [inline src/index.ts:81]: Can you please cleanup the TODO if we're not about to implement it? :slightly_smiling_face:

**Review by tumido — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 3 above)
Comment (schwesig) [reply src/index.ts]: done

File: src/index.ts [hunk 4]
```diff
@@ -72,6 +72,14 @@ export = (app: Probot) => {
     };
     await k8sApi.createNamespacedSecret(namespace,secret);
   });
+
+  // Respond to the GitHub app installation deleted
+  app.on("installation.deleted", async (context) =>{
+
+    var name = "peribolos-"+context.payload.installation.id;
```
[the remark this answers is not in the document]
Comment (schwesig) [reply src/index.ts]: done

**Review by tumido — APPROVED** [review]
/lgtm :rocket: :+1:
/approve

**Pull request merged**